deploy/src/crd.rs

```diff
@@ -0,0 +1,166 @@
+#![allow(dead_code)] // remove when it is implemented
+
+#[cfg(test)]
+use garde::Validate;
+use k8s_openapi::api::core::v1::{Container, PersistentVolumeClaim};
+use k8s_openapi::serde::{Deserialize, Serialize};
+use kube::CustomResource;
+use schemars::JsonSchema;
+
+// garde is for validation locally
+
+/// Current CRD `XineCluster` version
+pub(crate) type Cluster = v1::Cluster;
+
+// The `CustomResource` macro generates a struct `Cluster` that does not pass the clippy lint.
+#[allow(clippy::str_to_string)]
+#[allow(clippy::missing_docs_in_private_items)]
```

Comment:
Please add a comment to explain why these lints are allowed

Reply:
The explanation is on line 15.